helpers/kubernetes/cluster.go

```diff
@@ -208,6 +197,44 @@ func (c *Cluster) IsPodRunning(ctx context.Context, podName, namespace string) w
 	}
 }
 
+// IsPodDone returns a condition function that indicates whether the given pod has
+// terminated in some way or not. Use this for oneshot pods, i.e. pods launched by a Job
+// and expected to terminate.  The pod is considered terminated if __at least one__ of its
+// container is terminated. This condition takes into account that even a one-shot job may
+// contain a non-terminating container (linkerd!)
+
+func (c *Cluster) IsPodDone(ctx context.Context, podName, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := c.Kubectl.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cstatus := range pod.Status.ContainerStatuses {
+			if cstatus.State.Terminated != nil {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+}
+
+// IsPodFailed determines whether the given and terminated pod has failed. Use this for
+// oneshot pods, i.e. pods launched by a Job and expected to terminate. It is considered
+// failed if at least one of the terminated container reports "Error" as reason for the
+// termination.
+func (c *Cluster) IsPodFailed(ctx context.Context, podName, namespace string) (bool, error) {
```

Comment:
I think it's better to use the `Job` object instead. While trying out this branch, my staging failed because [I don't have linkerd installed](https://github.com/epinio/epinio/blob/main/scripts/prepare-environment-k3d.sh#L47). This cause the staging Job to fail with an error: [`curl: (7) Failed to connect to localhost port 4191: Connection refused`](https://github.com/epinio/epinio/blob/f25fbbcee666ec60e05564b3e380fe60556c1736/internal/api/v1/application/stage.go#L364-L365). When I tried to stage again, I couldn't because of: 

```
"error"="Bad Request: pipelinerun for image ID still running" "status"=400
```
(the error message need fixing but the essence of it is that the job is not considered failed yet although it has this status:

```
191 status:                                                                         
192   conditions:                                                                   
193   - lastProbeTime: "2022-02-02T07:51:59Z"                                       
194     lastTransitionTime: "2022-02-02T07:51:59Z"                                  
195     message: Job has reached the specified backoff limit                        
196     reason: BackoffLimitExceeded                                                
197     status: "True"                                                              
198     type: Failed                                                                
199   failed: 1  
```

Reply:
Have you checked that the failure of the `Pod` was registered in the `Job` state ? I did not see that. 

Will fix the left-over `pipelinerun` references. totally forgotten that term, just looked for `tekton`.

Reply:
Done the left-over pipeline references. No work on the `still running` yet.